pkg/providers/v1/aws.go

```diff
@@ -928,6 +928,19 @@ func extractIPv4NodeAddresses(instance *ec2.Instance) ([]v1.NodeAddress, error)
 
 	addresses := []v1.NodeAddress{}
 
+	// sort by device index so that the first address added to the addresses list is from the first (primary) device
+	sort.Slice(instance.NetworkInterfaces, func(i, j int) bool {
```

Comment:
Can we write a unit test for this to confirm that the ordering is what we intend?

Reply:
I could give it a try, but I didn't do so already because I couldn't find much in the way of existing tests here.  As a result, I don't have any good examples to base my work on.  Did I overlook something?

Reply:
That's good information to have. I don't know that it necessarily needs to be a strict blocker, but I don't think it would be too crazy.

I don't see any asynchronous calls here, so it should act entirely on the inputs. You could separate this out into another method to really narrow down the testing, but shouldn't be much more scope to leave as is. [Here's a somewhat similar sort of test](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v1/aws_test.go#L3120-L3177).

Reply:
Thanks for the link.  I was looking in the wrong place for tests.  That's not surprising since I'm very green with golang.  I'll try to add some tests tomorrow and push them up.

Reply:
I finally managed to understand how the fakes work here, but I don't see any tests for this functionality.  I see some tests for Fargate instances, but they don't use the NetworkInterfaces list and rely solely on the PrivateDnsName and PrivateIpAddress fields of the instance.  Shouldn't that be sufficient for non-Fargate instances as well?  In other words, why does this controller attempt to list all addresses for all network interfaces for a node rather than just the primary address and DNS name?

Reply:
@mmerkes, do you know if we can simplify this whole thing and just use the primary address and DNS name of the nodes rather than iterate over all attached addresses?

Reply:
A test has been added as requested.